File: Beecrowd_Parte2/Empurrando_Blocos_1874.java
import java.util.Scanner;

public class Empurrando_Blocos_1874 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (sc.hasNext()) {
            int H = sc.nextInt();
            int P = sc.nextInt();
            int F = sc.nextInt();

            if (H == 0 && P == 0 && F == 0) {
                break;
            }

            int[][] pilhas = new int[H][P];
            int[] alturas = new int[P];
            int[] fila = new int[F];

            // Ler as pilhas
            for (int i = 0; i < H; i++) {
                for (int j = 0; j < P; j++) {
                    pilhas[i][j] = sc.nextInt();
                    if (pilhas[i][j] == 1) {
                        alturas[j]++;
                    }
                }
            }

            // Ler a fila
            for (int i = 0; i < F; i++) {
                fila[i] = sc.nextInt();
            }

            // Processar a fila de blocos
            for (int i = 0; i < F; i++) {
                int bloco = fila[i];
                for (int j = P - 1; j >= 0; j--) {
                    if (alturas[j] < H) {
                        pilhas[alturas[j]][j] = bloco;
                        alturas[j]++;
                        break;
                    }
                }
            }

            // Imprimir o resultado
            for (int i = 0; i < H; i++) {
                for (int j = 0; j < P; j++) {
                    if (j > 0) System.out.print(" ");
                    System.out.print(pilhas[i][j]);
                }
                System.out.println();
            }
            if (sc.hasNext()) {
                System.out.println();
            }
        }

        sc.close();
    }
}

File: Beecrowd_Parte2/Empurrando_Blocos_1874.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_H 105
#define MAX_P 105
#define MAX_F 1005

int pilhas[MAX_H][MAX_P];
int alturas[MAX_P];
int fila[MAX_F];

void processaEntrada(int H, int P, int F) {
    // Ler as pilhas
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < P; j++) {
            scanf("%d", &pilhas[i][j]);
        }
    }

    // Ler a fila
    for (int i = 0; i < F; i++) {
        scanf("%d", &fila[i]);
    }

    // Calcular a altura atual de cada pilha
    for (int j = 0; j < P; j++) {
        alturas[j] = 0;
        for (int i = 0; i < H; i++) {
            if (pilhas[i][j] == 1) {
                alturas[j]++;
            }
        }
    }

    // Processar a fila de blocos
    for (int i = 0; i < F; i++) {
        int bloco = fila[i];
        for (int j = P - 1; j >= 0; j--) {
            if (alturas[j] < H) {
                pilhas[alturas[j]][j] = bloco;
                alturas[j]++;
                break;
            }
        }
    }

    // Imprimir o resultado
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < P; j++) {
            if (j > 0) printf(" ");
            printf("%d", pilhas[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int H, P, F;

    while (scanf("%d %d %d", &H, &P, &F) == 3) {
        if (H == 0 && P == 0 && F == 0) {
            break;
        }

        processaEntrada(H, P, F);
        printf("\n");
    }

    return 0;
}
